genesyscloud/journey_outcome_predictor/resource_genesyscloud_journey_outcome_predictor.go

```diff
@@ -0,0 +1,121 @@
+	}
+
+	return resources, nil
+}
+
+// createJourneyOutcomePredictor is used by the journey_outcome_predictor resource to create Genesys cloud journey outcome predictor
+func createJourneyOutcomePredictor(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	op := getJourneyOutcomePredictorProxy(sdkConfig)
+
+	predictorRequest := getPredictorFromResourceData(d)
+
+	predictor, err := op.createJourneyOutcomePredictor(ctx, &predictorRequest)
+	if err != nil {
+		return diag.Errorf("Failed to create predictor: %s", err)
+	}
+
+	d.SetId(*predictor.Id)
+	log.Printf("Created predictor %s", *predictor.Id)
+	return readJourneyOutcomePredictor(ctx, d, meta)
+}
+
+// readJourneyOutcomePredictor is used by the journey_outcome_predictor resource to read an journey outcome predictor from genesys cloud
+func readJourneyOutcomePredictor(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	op := getJourneyOutcomePredictorProxy(sdkConfig)
+
+	log.Printf("Reading predictor %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		_, respCode, getErr := op.getJourneyOutcomePredictorById(ctx, d.Id())
+		if getErr != nil {
+			if util.IsStatus404ByInt(respCode) {
+				return retry.RetryableError(fmt.Errorf("Failed to read predictor %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("Failed to read predictor %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceJourneyOutcomePredictor())
+
+		return cc.CheckState()
+	})
+}
+
+// deleteJourneyOutcomePredictor is used by the journey_outcome_predictor resource to delete an journey outcome predictor from Genesys cloud
+func deleteJourneyOutcomePredictor(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	op := getJourneyOutcomePredictorProxy(sdkConfig)
+
+	_, err := op.deleteJourneyOutcomePredictor(ctx, d.Id())
+	if err != nil {
+		return diag.Errorf("Failed to delete predictor %s: %s", d.Id(), err)
+	}
+
+	return util.WithRetries(ctx, 180*time.Second, func() *retry.RetryError {
+		_, respCode, err := op.getJourneyOutcomePredictorById(ctx, d.Id())
+
+		if err == nil {
+			return retry.NonRetryableError(fmt.Errorf("Error deleting predictor %s: %s", d.Id(), err))
+		}
+		if util.IsStatus404ByInt(respCode) {
+			// Success  : External contact deleted
+			log.Printf("Deleted predictor %s", d.Id())
+			return nil
+		}
+
+		return retry.RetryableError(fmt.Errorf("Predictor %s still exists", d.Id()))
+	})
+}
+
+// getPredictorFromResourceData maps data from schema ResourceData object to a platformclientv2.Externalcontact
+func getPredictorFromResourceData(d *schema.ResourceData) platformclientv2.Outcomepredictorrequest {
```

Comment:
Honestly this functionality can probably put inside `createJourneyOutcomePredictor`. Functions like this are only really needed when we are creating the same object in the create and update function but since there's no update here it's not really necessary